### PR TITLE
feat:updated ts support and azure eu support [ECO-2972]

### PR DESCRIPTION
**Pull request by Amitkanswal**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=DjWBagJ)
### Changes-
- TypeScript updated
- SDK initialization updated
- Library updated
- Azure EU support
### Story-[ECO-2972]

### CODE REVIEW

File: src/components/FooterContent.vue [hunk 1]
```diff
@@ -49,55 +50,44 @@
...
-      this.data = response[0];
-      this.$store.dispatch('setFooter', response[0]);
+      response.navigation.link = navFooterList;
+      this.data = response;
+      this.$store.dispatch('setFooter', response);
     },
   },
   mounted() {
     onEntryChange(() => {
       if (process.env.VUE_APP_CONTENTSTACK_LIVE_PREVIEW === 'true') {
```
Comment (abhishek305) [inline src/components/FooterContent.vue:93]: suggestion : destruct process.env

File: src/plugins/utils.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      '_'
+    ) as keyof typeof Region;
+  }
+  return Region[region];
+};
+// set LivePreview config
+const setLivePreviewConfig = (): LivePreview => {
+  if (!isLpConfigValid())
+    throw new Error(
+      'Your LP config is set to true. Please make you have set all required LP config in .env'
```
Comment (abhishek305) [inline src/plugins/utils.ts:46]: is this sentence verified?

**Review by abhishek305 — APPROVED** [review]
minor comments rest looks good

File: src/plugins/utils.ts (hunk 2 above)
Comment (Amitkanswal) [reply src/plugins/utils.ts:46]: by?

**Review by surajcontentstack — APPROVED** [review]

File: src/components/FooterContent.vue (hunk 1 above)
Comment (Amitkanswal) [reply src/components/FooterContent.vue:93]: resolved

**Pull request merged**